map_test.go

```diff
@@ -1090,6 +1145,109 @@ func TestMapIterate(t *testing.T) {
 	}
 }
 
+func TestMapDrain(t *testing.T) {
+	t.Run(Hash.String(), func(t *testing.T) {
+		hash, err := NewMap(&MapSpec{
+			Type:       Hash,
+			KeySize:    5,
+			ValueSize:  4,
+			MaxEntries: 2,
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer hash.Close()
+
+		if err := hash.Put("hello", uint32(21)); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := hash.Put("world", uint32(42)); err != nil {
+			t.Fatal(err)
+		}
+
+		var (
+			key   string
+			value uint32
+			keys  []string
+		)
+
+		entries := hash.Drain()
+		for entries.Next(&key, &value) {
+			keys = append(keys, key)
+		}
+
+		if err := entries.Err(); err != nil {
+			t.Fatal(err)
+		}
+
+		sort.Strings(keys)
+
+		if n := len(keys); n != 2 {
+			t.Fatal("Expected to get 2 keys, have", n)
+		}
+		if keys[0] != "hello" {
+			t.Error("Expected index 0 to be hello, got", keys[0])
+		}
+		if keys[1] != "world" {
+			t.Error("Expected index 1 to be hello, got", keys[1])
+		}
+	})
+
+	t.Run(Queue.String(), func(t *testing.T) {
```

Comment:
What about stack?

Reply:
> What about stack?

As suggested in https://github.com/cilium/ebpf/pull/1349#discussion_r1502624529, probably Stack vs Queue it's a matter of implementation in kernel space, don't know if it's really needed for us to test it here.